### PR TITLE
i18n(fr): Update components, configuration, sidebar and customization files

### PR DESCRIPTION
**Pull request by thomasbnt**
#### Description

Update some files in French language.

### CODE REVIEW

File: docs/src/content/docs/fr/guides/components.mdx [hunk 1]
```diff
@@ -205,3 +205,49 @@ Le code ci-dessus génère ce qui suit sur la page :
 Une liste de toutes les icônes disponibles est affichée ci-dessous avec leurs noms associés. Cliquez sur une icône pour copier le code du composant.
 
 <IconsList />
+
+### Code
+
+Utilisez le composant `<Code>` pour rendre le code en évidence syntaxique lorsque l'utilisation d'un [bloc de code Markdown](/fr/guides/authoring-content/#blocs-de-code) n'est pas possible, par exemple, pour rendre des données provenant de sources externes comme des fichiers, des bases de données ou des API.
```
Comment (HiDeoo) [inline docs/src/content/docs/fr/guides/components.mdx:211]:
```suggestion
Utilisez le composant `<Code>` pour afficher du code avec coloration syntaxique lorsque l'utilisation d'un [bloc de code Markdown](/fr/guides/authoring-content/#blocs-de-code) n'est pas possible, par exemple, pour afficher des données provenant de sources externes comme des fichiers, des bases de données ou des API.
```

- I personally have never heard of "évidence syntaxique" before and I cannot find any reference to it. We have also been [using](https://starlight.astro.build/fr/guides/authoring-content/#blocs-de-code) "coloration syntaxique" in the past so I would suggest to use that instead.
- When translating the [overrides references](https://starlight.astro.build/reference/overrides/) page which contains a lot of time the verb "render", we decided that "rendre" does not really fit the context and is even used a lot for that matter. We decided to use "afficher" instead when possible.

File: docs/src/content/docs/fr/guides/components.mdx [hunk 2]
```diff
@@ -205,3 +205,49 @@ Le code ci-dessus génère ce qui suit sur la page :
...
+
+Voir la documentation [Expressive Code "Code Component"](https://expressive-code.com/key-features/code-component/) pour plus de détails sur les fonctions que `<Code>` supporte.
```
Comment (HiDeoo) [inline docs/src/content/docs/fr/guides/components.mdx:213]:
```suggestion
Consultez la documentation [du « composant Code » d'Expressive Code](https://expressive-code.com/key-features/code-component/) pour plus de détails sur les props que `<Code>` supporte.
```

- I switched to French quotation marks and translated the link
- `props` is something that the Astro Docs French translation [does not translate](https://docs.astro.build/fr/core-concepts/astro-components/#props-de-composant) so we usually do the same for consistency.

File: docs/src/content/docs/fr/guides/components.mdx [hunk 3]
```diff
@@ -205,3 +205,49 @@ Le code ci-dessus génère ce qui suit sur la page :
...
+
+```mdx
+# src/content/docs/example.mdx
```
Comment (HiDeoo) [inline docs/src/content/docs/fr/guides/components.mdx:216]:
```suggestion
# src/content/docs/exemple.mdx
```

We usually try to translate file name when this make sense, e.g. the first example on [this page](https://starlight.astro.build/fr/reference/frontmatter/).

File: docs/src/content/docs/fr/guides/components.mdx [hunk 4]
```diff
@@ -205,3 +205,49 @@ Le code ci-dessus génère ce qui suit sur la page :
...
+
+import { Code } from '@astrojs/starlight/components';
+
+export const exampleCode = `console.log('Cela peut provenir d'un fichier ou d'un CMS !');`;
```
Comment (HiDeoo) [inline docs/src/content/docs/fr/guides/components.mdx:220]:
```suggestion
export const exampleCode = `console.log("Cela peut provenir d'un fichier ou d'un CMS !");`;
```

The quotes needs to be switched due to the use of `'` inside the quote.

File: docs/src/content/docs/fr/guides/components.mdx [hunk 5]
```diff
@@ -205,3 +205,49 @@ Le code ci-dessus génère ce qui suit sur la page :
...
+export const fileName = 'example.js';
+export const highlights = ["file", "CMS"];
```
Comment (HiDeoo) [inline docs/src/content/docs/fr/guides/components.mdx:222]:
```suggestion
export const highlights = ["fichier", "CMS"];
```

This needs to match the output

File: docs/src/content/docs/fr/guides/components.mdx [hunk 6]
```diff
@@ -205,3 +205,49 @@ Le code ci-dessus génère ce qui suit sur la page :
...
+
+<Code code={exampleCode} lang="js" title={fileName} mark={highlights} />
+```
+
+Le code ci-dessus génère la page suivante :
```
Comment (HiDeoo) [inline docs/src/content/docs/fr/guides/components.mdx:227]:
```suggestion
Le code ci-dessus génère ce qui suit sur la page :
```

This technically does not generate a page so I changed to something closer to the original text.

File: docs/src/content/docs/fr/guides/components.mdx [hunk 7]
```diff
@@ -205,3 +205,49 @@ Le code ci-dessus génère ce qui suit sur la page :
...
+
+import { Code } from '@astrojs/starlight/components';
+
+export const exampleCode = `console.log('Cela peut provenir d'un fichier ou d'un CMS !');`;
```
Comment (HiDeoo) [inline docs/src/content/docs/fr/guides/components.mdx:231]:
```suggestion
export const exampleCode = `console.log("Cela peut provenir d'un fichier ou d'un CMS !");`;
```

Same as above

File: docs/src/content/docs/fr/guides/components.mdx [hunk 8]
```diff
@@ -205,3 +205,49 @@ Le code ci-dessus génère ce qui suit sur la page :
...
+export const fileName = 'example.js';
+export const highlights = ["file", "CMS"];
```
Comment (HiDeoo) [inline docs/src/content/docs/fr/guides/components.mdx:233]:
```suggestion
export const highlights = ["fichier", "CMS"];
```

File: docs/src/content/docs/fr/guides/components.mdx [hunk 9]
```diff
@@ -205,3 +205,49 @@ Le code ci-dessus génère ce qui suit sur la page :
...
+
+Voir la documentation [Expressive Code "Code Component"](https://expressive-code.com/key-features/code-component/) pour plus de détails sur les fonctions que `<Code>` supporte.
+
+```mdx
+# src/content/docs/example.mdx
+
+import { Code } from '@astrojs/starlight/components';
+
+export const exampleCode = `console.log('Cela peut provenir d'un fichier ou d'un CMS !');`;
+export const fileName = 'example.js';
```
Comment (HiDeoo) [inline docs/src/content/docs/fr/guides/components.mdx:221]:
```suggestion
export const fileName = 'exemple.js';
```

File: docs/src/content/docs/fr/guides/components.mdx [hunk 10]
```diff
@@ -205,3 +205,49 @@ Le code ci-dessus génère ce qui suit sur la page :
...
+
+<Code code={exampleCode} lang="js" title={fileName} mark={highlights} />
+```
+
+Le code ci-dessus génère la page suivante :
+
+import { Code } from '@astrojs/starlight/components';
+
+export const exampleCode = `console.log('Cela peut provenir d'un fichier ou d'un CMS !');`;
+export const fileName = 'example.js';
```
Comment (HiDeoo) [inline docs/src/content/docs/fr/guides/components.mdx:232]:
```suggestion
export const fileName = 'exemple.js';
```

File: docs/src/content/docs/fr/guides/components.mdx [hunk 11]
```diff
@@ -205,3 +205,49 @@ Le code ci-dessus génère ce qui suit sur la page :
...
+export const highlights = ["file", "CMS"];
+
+<Code code={exampleCode} lang="js" title={fileName} mark={highlights} />
+
+#### Code importé
+
+Utilisez [le suffixe d'importation `?raw` de Vite](https://vitejs.dev/guide/assets#importing-asset-as-string) pour importer n'importe quel fichier de code sous forme de chaîne de caractères.
```
Comment (HiDeoo) [inline docs/src/content/docs/fr/guides/components.mdx:239]:
```suggestion
Utilisez [le suffixe d'import `?raw` de Vite](https://vitejs.dev/guide/assets#importing-asset-as-string) pour importer n'importe quel fichier de code sous forme de chaîne de caractères.
```

Any thoughts on using just ["import"](https://fr.wiktionary.org/wiki/import) here?

File: docs/src/content/docs/fr/guides/components.mdx [hunk 12]
```diff
@@ -205,3 +205,49 @@ Le code ci-dessus génère ce qui suit sur la page :
...
+Vous pouvez ensuite passer cette chaîne importée au composant `<Code>` pour l'inclure dans votre page.
+
+```mdx title="src/content/docs/example.mdx" "?raw"
```
Comment (HiDeoo) [inline docs/src/content/docs/fr/guides/components.mdx:242]:
```suggestion
```mdx title="src/content/docs/exemple.mdx" "?raw"
```

File: docs/src/content/docs/fr/guides/components.mdx [hunk 13]
```diff
@@ -205,3 +205,49 @@ Le code ci-dessus génère ce qui suit sur la page :
...
+import { Code } from '@astrojs/starlight/components';
+import importedCode from '/src/env.d.ts?raw';
+
+<Code code={importedCode} lang="ts" title="src/env.d.ts" />
+```
+
+Le code ci-dessus génère la page suivante :
```
Comment (HiDeoo) [inline docs/src/content/docs/fr/guides/components.mdx:249]:
```suggestion
Le code ci-dessus génère ce qui suit sur la page :
```

File: docs/src/content/docs/fr/guides/customization.mdx [hunk 14]
```diff
@@ -277,6 +277,25 @@ hero:
 ---
 ```
 
+### Désactiver la page 404 par défaut
+
+Si votre projet nécessite une mise en page 404 entièrement personnalisée, vous pouvez créer une route `src/pages/404.astro` et définir l'option de configuration [`disable404Route`](/fr/reference/configuration/#disable404route) pour désactiver la route par défaut de Starlight :
```
Comment (HiDeoo) [inline docs/src/content/docs/fr/guides/customization.mdx:282]:
```suggestion
Si votre projet nécessite une mise en page entièrement personnalisée pour votre page 404, vous pouvez créer une route `src/pages/404.astro` et définir l'option de configuration [`disable404Route`](/fr/reference/configuration/#disable404route) pour désactiver la route par défaut de Starlight :
```

A bit more verbose than the original version but I found "mise en page 404" pretty confusing.

File: docs/src/content/docs/fr/guides/customization.mdx [hunk 15]
```diff
@@ -277,6 +277,25 @@ hero:
...
+
+```js {9}
+// astro.config.mjs
+import { defineConfig } from 'astro/config';
+import starlight from '@astrojs/starlight';
+
+export default defineConfig({
+	integrations: [
+		starlight({
+			title: 'Docs avec 404 personnalisé',
```
Comment (HiDeoo) [inline docs/src/content/docs/fr/guides/customization.mdx:292]:
```suggestion
			title: 'Documentation avec 404 personnalisée',
```

File: docs/src/content/docs/fr/guides/sidebar.mdx [hunk 16]
```diff
@@ -141,7 +138,8 @@ La configuration ci-dessus génère la barre latérale suivante :
 
 Starlight peut générer automatiquement un groupe dans votre barre latérale en fonction d'un répertoire de votre documentation.
 Cela est utile lorsque vous ne souhaitez pas entrer manuellement chaque élément de la barre latérale dans un groupe.
-Les pages seront triées par ordre alphabétique en fonction du nom de fichier par défaut.
+
+Par défaut, les pages sont triées par ordre alphabétique selon le fichier [`slug`](/fr/reference/overrides/#slug).
```
Comment (HiDeoo) [inline docs/src/content/docs/fr/guides/sidebar.mdx:142]:
```suggestion
Par défaut, les pages sont triées par ordre alphabétique selon le [`slug`](/fr/reference/overrides/#slug) du fichier.
```

File: docs/src/content/docs/fr/guides/sidebar.mdx [hunk 17]
```diff
@@ -208,8 +203,7 @@ title: Ma page
 sidebar:
   # Définit une étiquette personnalisée pour le lien dans la barre latérale
   label: Étiquette personnalisée
-  # Définit un ordre personnalisé pour le lien (les nombres plus petits
-  # sont affichés plus haut)
+  # Définit un ordre personnalisé pour le lien (les nombres plus petits sont affichés plus haut)
```
Comment (HiDeoo) [inline docs/src/content/docs/fr/guides/sidebar.mdx:206]: This comment now requires scrolling to see the entire comment on desktop compared to the previous version. I think this should be reverted.

File: docs/src/content/docs/fr/guides/sidebar.mdx [hunk 18]
```diff
@@ -480,8 +474,7 @@ starlight({
 	sidebar: [
 		{
 			label: 'Guides',
-			// Rétracte le groupe et ses sous-groupes générés automatiquement
-			// par défaut.
+			// Rétracte le groupe et ses sous-groupes générés automatiquement par défaut.
```
Comment (HiDeoo) [inline docs/src/content/docs/fr/guides/sidebar.mdx:477]: Same as the previous one.

File: docs/src/content/docs/fr/reference/configuration.mdx [hunk 19]
```diff
@@ -516,6 +514,13 @@ Définit le délimiteur entre le titre de la page et le titre du site dans la ba
 Par défaut, chaque page a une balise `<title>` contenant `Titre de la page | Titre du site`.
 Par example, cette page a pour titre « Référence de configuration » et ce site a pour titre « Starlight », donc la balise `<title>` de cette page contient « Référence de configuration | Starlight ».
 
+### `disable404Route`
+
+**Type :** `boolean`
+**Par défaut :** `'false'`
```
Comment (HiDeoo) [inline docs/src/content/docs/fr/reference/configuration.mdx:520]:
```suggestion
**Type :** `boolean`  
**Par défaut :** `'false'`
```

We want the 2 informations on 2 lines.

File: docs/src/content/docs/fr/reference/configuration.mdx [hunk 20]
```diff
@@ -516,6 +514,13 @@ Définit le délimiteur entre le titre de la page et le titre du site dans la ba
...
+
+Désactive l'injection de la [page 404](https://docs.astro.build/fr/core-concepts/astro-pages/#custom-404-error-page) par défaut de Starlight. Pour utiliser une route personnalisée `src/pages/404.astro` dans votre projet, mettez cette option à `false`.
```
Comment (HiDeoo) [inline docs/src/content/docs/fr/reference/configuration.mdx:522]:
```suggestion
Désactive l'injection de la [page 404](https://docs.astro.build/fr/core-concepts/astro-pages/#page-derreur-404-personnalis%C3%A9e) par défaut de Starlight. Pour utiliser une route personnalisée `src/pages/404.astro` dans votre projet, définissez cette option à `false`.
```

**Review by HiDeoo — CHANGES_REQUESTED** [review]
Thanks for the translations :raised_hands:

I left quite a few comments and suggestions but a good part of them are pretty minor, tiny mistakes or stylistic choices to keep consistency with the rest of the translation.

One thing I noted tho is that this PR introduces a lot of changes which are unrelated to the new translation itself. If I had to guess I would say maybe you have an editor having its own formatting settings or something like that? This makes it a bit more difficult to review the PR but sometime, it also collapse some code example comments which were explicitly on multiple lines into a single line comment which could technically break some highlighting (this is not the case here) but it sometimes introduce a scroll bar in the code example which is not ideal.

I wonder if it wouldn't be easier in this PR to revert all the unwanted changes :thinking: Do you have any thoughts on why this happened?

**Comment by thomasbnt** [comment]
> One thing I noted tho is that this PR introduces a lot of changes which are unrelated to the new translation itself. If I had to guess I would say maybe you have an editor having its own formatting settings or something like that? This makes it a bit more difficult to review the PR but sometime, it also collapse some code example comments which were explicitly on multiple lines into a single line comment which could technically break some highlighting (this is not the case here) but it sometimes introduce a scroll bar in the code example which is not ideal.

Hello @HiDeoo ! I've edited these lines to be consistent with the total number of lines in the original files

File: docs/src/content/docs/fr/guides/components.mdx (hunk 2 above)
Comment (thomasbnt) [reply docs/src/content/docs/fr/guides/components.mdx:213]: Alright! Thanks for this clarification!

File: docs/src/content/docs/fr/guides/components.mdx (hunk 1 above)
Comment (thomasbnt) [reply docs/src/content/docs/fr/guides/components.mdx:211]: "évidence syntaxique" 
I thought it was weird, but I messed it up and couldn't find the right words at the time, so I'm glad you noticed! 🙏🏼

File: docs/src/content/docs/fr/guides/components.mdx (hunk 3 above)
Comment (thomasbnt) [reply docs/src/content/docs/fr/guides/components.mdx:216]: okay! I didn't find it necessary to translate even the example files... in the examples. But if I have to, I'll merge the whole thing! Thanks!

File: docs/src/content/docs/fr/guides/components.mdx (hunk 11 above)
Comment (thomasbnt) [reply docs/src/content/docs/fr/guides/components.mdx:239]: yeah we can also use "import" in this text!

File: docs/src/content/docs/fr/guides/customization.mdx (hunk 14 above)
Comment (thomasbnt) [reply docs/src/content/docs/fr/guides/customization.mdx:282]: Seems more logic!

File: docs/src/content/docs/fr/reference/configuration.mdx [hunk 21]
```diff
@@ -516,6 +514,13 @@ Définit le délimiteur entre le titre de la page et le titre du site dans la ba
 Par défaut, chaque page a une balise `<title>` contenant `Titre de la page | Titre du site`.
 Par example, cette page a pour titre « Référence de configuration » et ce site a pour titre « Starlight », donc la balise `<title>` de cette page contient « Référence de configuration | Starlight ».
 
+### `disable404Route`
+
+**Type :** `boolean`  
+**Par défaut :** `'false'`
```
Comment (HiDeoo) [inline docs/src/content/docs/fr/reference/configuration.mdx:520]:
```suggestion
**Par défaut :** `false`
```

**Review by HiDeoo — CHANGES_REQUESTED** [review]
Thanks for handling all the changes 🙌 

I added one last tiny suggestion which is a follow-up to https://github.com/withastro/starlight/pull/1418 and also, would you be able to remove the `.idea` folder which was added in the more recent commits? I think it's editor preferences?

After that, I think we should be good to merge 🚀

File: docs/src/content/docs/fr/reference/configuration.mdx [hunk 22]
```diff
@@ -516,6 +514,13 @@ Définit le délimiteur entre le titre de la page et le titre du site dans la ba
...
+
+Désactive l'injection de la [page 404](https://docs.astro.build/fr/core-concepts/astro-pages/#page-derreur-404-personnalis%C3%A9e) par défaut de Starlight. Pour utiliser une route personnalisée `src/pages/404.astro` dans votre projet, définissez cette option à `false`.
```
Comment (HiDeoo) [inline docs/src/content/docs/fr/reference/configuration.mdx:522]:
```suggestion
Désactive l'injection de la [page 404](https://docs.astro.build/fr/core-concepts/astro-pages/#page-derreur-404-personnalis%C3%A9e) par défaut de Starlight. Pour utiliser une route personnalisée `src/pages/404.astro` dans votre projet, définissez cette option à `true`.
```

One *last* change to be updated with the changes of https://github.com/withastro/starlight/pull/1431

**Comment by thomasbnt** [comment]
Thanks for your reviews!
> would you be able to remove the `.idea` folder which was added in the more recent commits? I think it's editor preferences?
Oh... him again... sorry arg...

**Review by HiDeoo — APPROVED** [review]
Thanks for the updates, looks perfect to me 🌟

**Pull request merged**